src/app/cube/browse/components/filter/filter.component.ts

```diff
@@ -1,70 +1,309 @@
   }
 
   /**
-   * Emits to parent component signal to remove filter from specified component
-   * @param category category that target filter is in
-   * @param filter filter to be removed
+   * Sets each filter to inactive in a filter category
+   *
+   * @param filters The array of filters
    */
-  removeFilter(category: string, filter: string) {
-    this.remove.emit({category, filter});
+  private clearFilterCategory(filters: { active: boolean, name: string, value: string, tip?: string }[]) {
+    filters.forEach(filter => filter.active = false);
+  }
+
+  /**
+   * Formats the filter query to search/filter
+   */
+  sendFilterChanges() {
+    const query = {};
+
+    // Creates the query
+    this.checkFilter('collection', this.collectionFilter.filters, query);
+    this.checkFilter('length', this.lengthFilter.filters, query);
+    this.checkFilter('level', this.levelFilter.filters, query);
+    this.checkFilter('fileTypes', this.materialFilter.filters, query);
+    this.checkFilter('topics', this.topicFilter.filters, query);
+    this.checkFilter('guidelines', this.frameworkFilter.filters, query);
+    if (this.guidelineFilter && this.guidelineFilter.length > 0) {
+      query['standardOutcomes'] = this.guidelineFilter;
+    }
+
+    // Emits changes
+    this.changed.emit(query);
+  }
+
+  /**
+   * Appends the filter query with the given filter category if it exists
+   *
+   * @param category the filter category to append
+   * @param filters The filters to append to the category
+   * @param query The query object to emit
+   */
+  private checkFilter(category: string, filters: { active: boolean, name: string, value: string, tip?: string }[], query: any) {
+    const f = filters.filter(filter => filter.active);
+    if (f && f.length > 0) {
+      query[category] = f.map(filter => filter.value);
+    }
+  }
+
+  /**
+   * Registers a change in the filter, used for debounce-ing the filter selection
+   */
+  registerChange() {
+    this.filterChanged$.next();
   }
-}
 
-// The title of the FilterSection is the title that appears in the UI.
-// The name of the FilterSection is the name that is sent to the backend for the specific filter.
-// Determining the title here will allow us to change the title without changing backend logic for the name of the specific filter.
-export interface FilterSection {
-  title: string;
-  name: string;
-  type: 'select-many' | 'select-one' | 'dropdown-one' | 'custom';
-  canSearch?: boolean;
-  values?: {
-    name: string;
-    value?: string;
-    active?: boolean;
-    toolTip?: string;
-  }[];
-  template?: TemplateRef<any>;
+  /**
+   * Gets the collection filters
+   */
+  async getCollectionFilters() {
+    const collections = await this.collectionService.getCollections();
+    this.collectionFilter = {
+      section: 'Collection',
+      filters: collections.map(collection => ({
+        name: collection.name,
+        value: collection.abvName || collection.name,
```

Comment:
Why is the abvName or name?

Reply:
I must have gotten fullname confused with name.  This should just be abvName.